packages/ui/src/components/Checkbox/index.tsx

```diff
@@ -374,11 +356,11 @@ export const Checkbox = forwardRef(
             type="checkbox"
             aria-invalid={!!error}
             aria-describedby={error ? `${localId}-hint` : undefined}
-            aria-checked={state === 'indeterminate' ? 'mixed' : state}
+            aria-checked={checked === 'indeterminate' ? 'mixed' : checked}
             aria-label={ariaLabel}
-            checked={state === 'indeterminate' ? false : state}
+            checked={checked === true ? checked : undefined}
             inputSize={size ?? theme.sizing['300']}
-            onChange={onLocalChange}
+            onChange={onChange}
```

Comment:
should we keep logic `onChange={progress ? onChange : undefined}` ? I guess not but a reminder just in case, may be we could disabled checkbox instead when `inProgress`

Reply:
done with disabled={disabled || progress}